extensions/builder.md

```diff
@@ -0,0 +1,180 @@
+# Builder Specification <!-- omit in toc -->
+
+This document specified the artifact format for a builder.
+
+A builder is an OCI image that provides a distributable build environment.
+
+A [platform][platform-spec] supporting the builder extension specification SHOULD allow users to configure the build environment with a provided builder.
+
+## Table of Contents <!-- omit in toc -->
+- [General Requirements](#general-requirements)
+  - [Builder API Version](#builder-api-version)
+  - [File/Directories](#filedirectories)
+  - [Environment Variables](#environment-variables)
+  - [Labels](#labels)
+- [Data Format](#data-format)
+  - [Labels](#labels-1)
+    - [`io.buildpacks.builder.metadata` (JSON)](#iobuildpacksbuildermetadata-json)
+    - [`io.buildpacks.buildpack.order` (JSON)](#iobuildpacksbuildpackorder-json)
+    - [`io.buildpacks.buildpack.layers` (JSON)](#iobuildpacksbuildpacklayers-json)
+    - [`io.buildpacks.lifecycle.apis` (JSON)](#iobuildpackslifecycleapis-json)
+
+## General Requirements
+A **builder** is an image that MUST contain an implementation of the lifecycle, and build-time environment, and MAY contain buildpacks. Platforms SHOULD use builders to ease the build process.
```

Comment:
Is it Platforms "SHOULD" or "MAY" use builders to ease the build process?

Reply:
I think it's MAY, but happy to change it. A platform may decide to just use the lifecycle and stitch everything together themselves. It is harder to have interoperability with builders from other providers ,but that's a design decision for the platform to make.